src/runtime_src/xocl/core/device.cpp

```diff
@@ -221,7 +221,8 @@ init_scheduler(xocl::device* device)
     throw xocl::error(CL_INVALID_PROGRAM,"Cannot initialize MBS before program is loadded");
 
   auto axlf = device->get_axlf();
-  if (is_sw_emulation()) {
+  bool isKdsSwEmu = xrt_core::config::get_flag_kds_sw_emu();
+  if (is_sw_emulation() && !isKdsSwEmu) {
```

Comment:
Remove the iskdsswemu variable and fold into if condition.

Reply:
added xrt_core::config::get_flag_kds_sw_emu() to if condition